File: backend/models/Medication.js
const mongoose = require('mongoose');

const medicationSchema = new mongoose.Schema({
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  name: {
    type: String,
    required: true
  },
  dosage: {
    type: String,
    required: true
  },
  frequency: {
    type: String,
    required: true,
    // ✅ ADDED 'daily', 'weekly', 'twice_daily' to this list
    enum: ['daily', 'weekly', 'twice_daily', 'thrice', 'twice', 'once']
  },
  times: [{
    type: String,
    required: true
  }],
  startDate: {
    type: Date,
    default: Date.now
  },
  endDate: Date,
  notes: String,
  isActive: {
    type: Boolean,
    default: true
  }
}, {
  timestamps: true
});

module.exports = mongoose.model('Medication', medicationSchema);

File: medicine_reminder_app/lib/models/medication.dart
class Medication {
  final String id;
  final String userId;
  final String name;
  final String dosage;
  final String frequency;
  final List<String> times;
  final DateTime startDate;
  final DateTime? endDate;
  final String? notes;
  final bool isActive;

  Medication({
    required this.id,
    required this.userId,
    required this.name,
    required this.dosage,
    required this.frequency,
    required this.times,
    required this.startDate,
    this.endDate,
    this.notes,
    this.isActive = true,
  });

  factory Medication.fromJson(Map<String, dynamic> json) {
    return Medication(
      id: json['_id'], // MongoDB uses '_id'
      userId: json['userId'],
      name: json['name'],
      dosage: json['dosage'],
      frequency: json['frequency'],
      // Ensure 'times' is treated as a list of strings
      times: List<String>.from(json['times']),
      startDate: DateTime.parse(json['startDate']),
      endDate: json['endDate'] != null ? DateTime.parse(json['endDate']) : null,
      notes: json['notes'],
      isActive: json['isActive'] ?? true,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'dosage': dosage,
      'frequency': frequency,
      'times': times,
      'startDate': startDate.toIso8601String(),
      'endDate': endDate?.toIso8601String(),
      'notes': notes,
    };
  }
}